source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -284,6 +280,40 @@ void EnvoyQuicClientStream::OnInitialHeadersComplete(bool fin, size_t frame_len,
       got_304_response_ = true;
     }
   }
+#else
+  // Extended CONNECT to H/1 upgrade transformation has moved to UHV
+  // In Envoy, both upgrade requests and extended CONNECT requests are
+  // represented as their HTTP/1 forms, regardless of the HTTP version used.
+  // Therefore, these need to be transformed into their HTTP/1 form.
+
+  // in UHV mode the :status header at this point can be malformed, as it is validated
+  // later on in the response_decoder_.decodeHeaders() call.
+  // Account for this here.
+  if (optional_status.has_value()) {
+    const uint64_t status = optional_status.value();
+    if (Http::CodeUtility::is1xx(status)) {
+      // These are Informational 1xx headers, not the actual response headers.
```

Comment:
Does this need to check enumToInt(Http::Code::SwitchingProtocols), too?

Reply:
`101 Switching protocols` is invalid for H/2 and H/3, since they do not support upgrades, and have the extended CONNECT which uses 200 status.

Reply:
Yeah, agreed. But I'm wondering if we need to detect this code and explicitly error out, since it's forbidden. Do we have an integration test for `101 Switching protocols` over H3 (with and without UHV)?

Reply:
This is a good question. The standard does not say that it is malformed, only that it is not supported https://www.rfc-editor.org/rfc/rfc9113#name-the-upgrade-header-field . I'm unsure if it needs to be treated as a protocol error by Envoy. Also not sure what effect it will have on H/3 by calling set_headers_decompressed(false); 

Thanks for pointing this out. I've filed #29071 (and left comment in code) since it is a bigger issue that needs separate look.